contrib/pkg/awstagdeprovision/awstagdeprovision.go

```diff
@@ -266,6 +267,106 @@ func deleteLBs(vpc *ec2.Vpc, awsSession *session.Session, logger log.FieldLogger
 	return false
 }
 
+// filterV2LBsByVPC will find all the load balancers in the provided list that are under the provided VPC
+func filterV2LBsByVPC(lbs []*elbv2.LoadBalancer, vpc *ec2.Vpc, logger log.FieldLogger) []*elbv2.LoadBalancer {
+	filteredLBs := []*elbv2.LoadBalancer{}
+
+	for _, lb := range lbs {
+		if *lb.VpcId == *vpc.VpcId {
+			filteredLBs = append(filteredLBs, lb)
+		}
+	}
+
+	return filteredLBs
+}
+
+func deleteV2LBs(vpc *ec2.Vpc, awsSession *session.Session, logger log.FieldLogger) bool {
+	logger.Debugf("Deleting V2 load balancers (%s)", *vpc.VpcId)
+	defer logger.Debugf("Exiting deleting V2 load balancers (%s)", *vpc.VpcId)
+	elbv2Client := elbv2.New(awsSession)
+
+	total := 0
+	filteredLBs := []*elbv2.LoadBalancer{}
+	if err := elbv2Client.DescribeLoadBalancersPages(&elbv2.DescribeLoadBalancersInput{}, func(results *elbv2.DescribeLoadBalancersOutput, lastPage bool) bool {
+		total += len(results.LoadBalancers)
+		filteredLBs = append(filteredLBs, filterV2LBsByVPC(results.LoadBalancers, vpc, logger)...)
+		return lastPage
+	}); err != nil {
+		logger.Errorf("Error listing V2 load balancers: %v", err)
+		return false
+	}
+	logger.Debugf("from %d total V2 load balancers, %d scheduled for deletion", total, len(filteredLBs))
+
+	if len(filteredLBs) == 0 {
+		// no items left to delete
+		// see if we can delete target groups.
+		return deleteTargetGroups(vpc, awsSession, logger)
```

Comment:
should we move this to a separate step in deleteVPC after we call delete V2LBs ? or this makes sense here too?

Reply:
> should we move this to a separate step in deleteVPC after we call delete V2LBs ?

Leaving it here works better with [this][1].  But as the `deleteVPC` logic stands, I don't think it matters either way.

[1]: https://github.com/openshift/hive/pull/62#discussion_r230140110

Reply:
keeping it as is keeps the deleteVPC simple, so keeping it as-is